scenario_player/utils/configuration/settings.py

```diff
@@ -41,8 +41,30 @@ class UDCTokenSettings(ConfigMapping):
     def __init__(self, loaded_yaml: dict):
         udc_settings = ((loaded_yaml.get("settings") or {}).get("services") or {}).get("udc") or {}
         super(UDCTokenSettings, self).__init__(udc_settings.get("token"))
+        self.validate(loaded_yaml)
         print(self.dict)
 
+    def validate(self, loaded_yaml):
+        self.assert_option(
+            self.max_funding >= self.balance_per_node,
+            "max_funding needs to be greater or equal to balance_per_node",
+        )
+
+        # Get the amount of tokens that will be minted (min_balance)
+        try:
+            min_balance = loaded_yaml["token"]["min_balance"]
+        except (KeyError, TypeError):
+            min_balance = DEFAULT_TOKEN_BALANCE_FUND
+        # The amount of Tokens minted need to be greater than the amount of tokens
+        # that will be deposited into the UDC
+        self.assert_option(
+            self.max_funding <= min_balance,
+            "Token.max_funding needs to be greater than the Deposit amount."
+            f"The Deposit amount is max({DEFAULT_TOKEN_BALANCE_FUND}, "
```

Comment:
This needs to be the actual value of `token.max_funding` - as as you've already pointed out, this check would be easier and cleaner to do on the `ScenarioYAML` class.